go/rep_addr_info.go

```diff
@@ -0,0 +1,101 @@
+package pb
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/multiformats/go-multiaddr"
+)
+
+// HostInfo is an interface that we can build a AddrInfo from.
+// It is a subset of libp2p/go-libp2p-core/host/#Host.
+type HostInfo interface {
+	ID() peer.ID
+	Addrs() []multiaddr.Multiaddr
+}
+
+//NewAddrInfoFromHost creates an AddrInfo from a HostInfo
+func NewAddrInfoFromHost(h HostInfo, gport int32) (*AddrInfo, error) {
+	a := &AddrInfo{
+		GrpcPort: gport,
+	}
+	a.SetID(h.ID())
+	for _, m := range h.Addrs() {
+		if err := a.AddMultiAddr(m); err != nil {
+			return nil, err
+		}
+	}
+	if len(a.AddrsBytes) == 0 { //TODO: remove this check when we support connection to a peer with ID only for replication feature
+		return nil, errors.New("host have no address")
+	}
+	return a, nil
+}
+
+// GetID returns the libp2p PeerID of AddrInfo,
+// unless any error is encountered during decoding.
+func (a *AddrInfo) GetID() (peer.ID, error) {
+	return peer.IDFromBytes(a.IdBytes)
+}
+
+// SetID sets the id_bytes field from libp2p PeerID
+func (a *AddrInfo) SetID(id peer.ID) {
+	bs, err := id.MarshalBinary()
+	if err != nil {
+		panic(err) // assert id.MarshalBinary never errors
+	}
+	a.IdBytes = bs
+}
+
+// GetMultiAddrs returns the location addresses of AddrInfo as Multiaddres.
+// If Multiaddr contains an ID part, it must match ID in AddrInfo.
+// If any error is encountered during decoding, the first error is returned,
```

Comment:
Small typo, `If any errors are encountered during decoding, the first error is returned`

Reply:
good call